pyogrio/raw.py

```diff
@@ -193,6 +233,25 @@ def write(
             "systems."
         )
 
+    # preprocess kwargs and split in dataset and layer creation options
+    dataset_kwargs = _preprocess_options_key_value(dataset_options or {})
+    layer_kwargs = _preprocess_options_key_value(layer_options or {})
+    if kwargs:
+        kwargs = _preprocess_options_key_value(kwargs)
+        dataset_option_names = _parse_options_names(
+            _get_driver_metadata_item(driver, "DMD_CREATIONOPTIONLIST")
+        )
+        layer_option_names = _parse_options_names(
+            _get_driver_metadata_item(driver, "DS_LAYER_CREATIONOPTIONLIST")
+        )
+        for k, v in kwargs.items():
+            if k in dataset_option_names:
+                dataset_kwargs[k] = v
+            elif k in layer_option_names:
+                layer_kwargs[k] = v
+            else:
+                raise ValueError(f"unrecognized option '{k}' for driver '{driver}'")
```

Comment:
Are we OK with raising an error here in case an option is not supported for the driver in question?

At the moment, we pass it through and the the GDAL warning message gets printed (so in effect we ignore the option).

For example:

```
In [18]: df = geopandas.read_file(geopandas.datasets.get_path("naturalearth_lowres"))

In [19]: pyogrio.write_dataframe(df, "test.geojson", spatial_index=False)
Warning 6: dataset test.geojson does not support layer creation option SPATIAL_INDEX
```

But with this change, we actually raise an error.

Reply:
OK, also added a test for this